src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -293,6 +296,26 @@ fun processMacroSimpleResolveVariants(element: RsMacroBodySimpleMatching, proces
     return simple.any { processor(it) }
 }
 
+fun processMetaItemResolveVariants(element: RsMetaItem, processor: RsResolveProcessor): Boolean {
+    if (element.parentOfType<RsMetaItem>()?.identifier?.text != "derive") return false
+    val traitName = element.identifier.text
+    val stdTrait = STD_DERIVE_TRAITS[traitName]
+    val project = element.project
+    val trait = StubIndex.getElements(
+        RsNamedElementIndex.KEY,
+        traitName,
+        project,
+        GlobalSearchScope.allScope(project),
+        RsNamedElement::class.java
+    ).find { e ->
```

Comment:
This is going to hit an index on every resolve, but I think we can easily add some caching here. What do you thing about adding something like `fun derivableTraits(module: Module): Collection<TraitItem>`, which would cache the results based on PsiModifictionCount?